tests/components/modern_forms/test_fan.py

```diff
@@ -0,0 +1,214 @@
+"""Tests for the Modern Forms fan platform."""
+from unittest.mock import patch
+
+from aiomodernforms import ModernFormsConnectionError
+
+from homeassistant.components.fan import (
+    ATTR_DIRECTION,
+    ATTR_PERCENTAGE,
+    DIRECTION_FORWARD,
+    DIRECTION_REVERSE,
+    DOMAIN as FAN_DOMAIN,
+    SERVICE_SET_DIRECTION,
+    SERVICE_SET_PERCENTAGE,
+)
+from homeassistant.components.modern_forms.const import (
+    ATTR_SLEEP_TIME,
+    DOMAIN,
+    SERVICE_CLEAR_FAN_SLEEP_TIMER,
+    SERVICE_SET_FAN_SLEEP_TIMER,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.components.modern_forms import init_integration
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_fan_state(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the creation and values of the Modern Forms lights."""
+    await init_integration(hass, aioclient_mock)
+
+    entity_registry = er.async_get(hass)
+
+    state = hass.states.get("fan.modernformsfan_fan")
+    print(state)
+    assert state
+    assert state.attributes.get(ATTR_PERCENTAGE) == 50
+    assert state.attributes.get(ATTR_DIRECTION) == DIRECTION_FORWARD
+    assert state.state == STATE_ON
+
+    entry = entity_registry.async_get("fan.modernformsfan_fan")
+    assert entry
+    assert entry.unique_id == "AA:BB:CC:DD:EE:FF_fan"
+
+
+async def test_change_state(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, caplog
+) -> None:
+    """Test the change of state of the Modern Forms segments."""
```

Comment:
Is "segments" correct here?

Reply:
Hold over from cut and paste.  cleaned up